framework/di/Container.php

```diff
@@ -524,7 +525,7 @@ public function resolveCallableDependencies(callable $callback, $params = [])
                     unset($params[$name]);
                 } elseif (!$associative && isset($params[0]) && $params[0] instanceof $className) {
                     $args[] = array_shift($params);
-                } elseif (Yii::$app->has($name) && ($obj = Yii::$app->get($name)) instanceof $className) {
+                } elseif (Yii::$app instanceof ServiceLocator && Yii::$app->has($name) && ($obj = Yii::$app->get($name)) instanceof $className) {
```

Comment:
`\Yii::$app` is essentially a global variable that can be easily overwritten and have any kind of value. Anyway, this obviously is subjective so I'll change it if that makes you happy.